example/grpc-example/main.go

```diff
@@ -0,0 +1,149 @@
+	// instagrpc.StreamServerInterceptor(sensor) to the list of server options when
+	// initializing the server
+	srv := grpc.NewServer(
+		grpc.UnaryInterceptor(instagrpc.UnaryServerInterceptor(sensor)),
+		grpc.StreamInterceptor(instagrpc.StreamServerInterceptor(sensor)),
+	)
+
+	// register an implementation of the service
+	pb.RegisterEchoServiceServer(srv, &Service{})
+
+	// start the server and listen to the incoming requests
+	go startServer(srv, *address+*port)
+
+	// create a channel to signal when all requests are send
+	done := make(chan struct{})
+
+	// start goroutine
+	go func() {
+		// create a new sensor for a client
+		sensor := instana.NewSensor("grpc-client")
+
+		for i := 1; i <= *nRequestsToSend; i++ {
+			log.Println("\n", "Send request #", i)
+			// init the client with Instana gRPC instrumentation and call the service
+			response := Call(context.Background(), sensor, *address+*port, testMessage)
+			log.Printf("Response << %s", response)
+		}
+
+		done <- struct{}{}
+	}()
+
+	<-done
+}
+
+// Call send pb.EchoRequest request with "message" to the "address" using generated grpc pb.EchoServiceClient client
+// and returns the message from the response
+func Call(ctx context.Context, sensor *instana.Sensor, address, message string) string {
+	conn, err := grpc.Dial(
+		address,
+		grpc.WithInsecure(),
+		grpc.WithBlock(),
+		// To instrument client calls add instagrpc.UnaryClientInterceptor(sensor) and
+		// instagrpc.StringClientInterceptor(sensor) to the DialOption list while dialing
+		// the GRPC server.
+		grpc.WithUnaryInterceptor(instagrpc.UnaryClientInterceptor(sensor)),
+		grpc.WithStreamInterceptor(instagrpc.StreamClientInterceptor(sensor)),
+	)
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
+	}
+
+	defer conn.Close()
+
+	// create a new service client
+	client := pb.NewEchoServiceClient(conn)
+
+	// The call should always start with an entry span (https://www.instana.com/docs/tracing/custom-best-practices/#start-new-traces-with-entry-spans)
+	// Normally this would be your HTTP/GRPC/message queue request span, but here we need to
+	// create it explicitly.
+	sp := sensor.Tracer().
+		StartSpan("client-call").
+		SetTag(string(ext.SpanKind), "entry")
+
+	// call the service
+	r, err := client.Echo(
+		// Create a context that holds the parent entry span and pass it to the GRPC call
+		instana.ContextWithSpan(ctx, sp),
+		&pb.EchoRequest{Message: message},
+	)
+
+	if err != nil {
+		log.Fatalf("error while echoing: %v", err)
+	}
+
+	return r.Message
+}
+
+// Service is used to implement pb.EchoService
+type Service struct {
+	pb.UnimplementedEchoServiceServer
+}
+
+// Echo implements pb.EchoService
+func (s *Service) Echo(ctx context.Context, in *pb.EchoRequest) (*pb.EchoReply, error) {
+	log.Printf("Request << %s", in.GetMessage())
+
+	// Extract the parent span and use its tracer to initialize any child spans to trace the calls
```

Comment:
Let's emphasize that this is purely optional and not required for minimal instrumentation

Reply:
This is done.